src/NetworkEntities.js

```diff
@@ -18,43 +18,50 @@ class NetworkEntities {
     NAF.log.write('Creating remote entity', entityData);
 
     var networkId = entityData.networkId;
+    var template = document.querySelector(entityData.template);
+    var clone = document.importNode(template.content, true);
+    var el = clone.firstElementChild;
 
-    var el = document.createElement('a-entity');
     el.setAttribute('id', 'naf-' + networkId);
-
-    this.initPosition(el, entityData.components);
-    this.initRotation(el, entityData.components);
-    this.addNetworkComponent(el, entityData);
-
     this.registerEntity(networkId, el);
 
     return el;
   }
 
+  setInitialComponents(entity, entityData) {
+    this.initPosition(entity, entityData.components);
+    this.initRotation(entity, entityData.components);
+    this.addNetworkComponent(entity, entityData);
+  }
+
   initPosition(entity, componentData) {
     var hasPosition = componentData.hasOwnProperty('position');
     if (hasPosition) {
       var position = componentData.position;
-      entity.setAttribute('position', position);
+
+      if (typeof position === "string") {
+        entity.setAttribute('position', position);
+      } else {
+        entity.setAttribute('position', `${position.x} ${position.y} ${position.z}`);
+      }
     }
   }
 
   initRotation(entity, componentData) {
     var hasRotation = componentData.hasOwnProperty('rotation');
     if (hasRotation) {
       var rotation = componentData.rotation;
-      entity.setAttribute('rotation', rotation);
+
+      if (typeof rotation === "string") {
+        entity.setAttribute('rotation', rotation);
+      } else {
+        entity.setAttribute('rotation', `${rotation.x} ${rotation.y} ${hasRotation.z}`);
+      }
     }
   }
 
   addNetworkComponent(entity, entityData) {
-    var networkData = {
-      template: entityData.template,
-      owner: entityData.owner,
-      networkId: entityData.networkId
-    };
-
-    entity.setAttribute('networked', networkData);
+    entity.setAttribute('networked', `template: ${entityData.template}; owner: ${entityData.owner}; networkId: ${entityData.networkId}`);
```

Comment:
I still really don't like this string serialization, but I'm unsure how to make everything work with the custom components lifecycle.

Reply:
In A-Frame, entity and component initialization is synchronous now (as of 0.5.0 ish); you can use the object form of `setAttribute` immediately upon creation of an entity.